src/hnsw.c

```diff
@@ -291,10 +291,36 @@ static float4 array_dist(ArrayType *a, ArrayType *b, usearch_metric_kind_t metri
         elog(ERROR, "expected equally sized arrays but got arrays with dimensions %d and %d", a_dim, b_dim);
     }
 
-    float4 *ax = (float4 *)ARR_DATA_PTR(a);
-    float4 *bx = (float4 *)ARR_DATA_PTR(b);
+    float4 *ax;
+    float4 *bx;
 
-    return usearch_dist(ax, bx, metric_kind, a_dim, usearch_scalar_f32_k);
+    bool convert_to_int = (metric_kind == usearch_metric_hamming_k);
+
+    if(convert_to_int) {
+        int32 *ax_int = (int32*) ARR_DATA_PTR(a);
+        int32 *bx_int = (int32*) ARR_DATA_PTR(b);
+
+        ax = (float4*) palloc(a_dim * sizeof(float4));
+        bx = (float4*) palloc(b_dim * sizeof(float4));
+
+        for (int i = 0; i < a_dim; i++) {
+            ax[i] = (float4) ax_int[i];
+            bx[i] = (float4) bx_int[i];
+        }
+    }
+    else {
+        ax = (float4*)ARR_DATA_PTR(a);
+        bx = (float4*)ARR_DATA_PTR(b);
+    }
+
+    float4 result = usearch_dist(ax, bx, metric_kind, a_dim, usearch_scalar_f32_k);
```

Comment:
I do not think this really solves the issue. We should not be measuring hamming distance of float arrays. We just get lucky here that when we cast ints to floats, they end up having the same byte arrangement and hamming distance calculation (an inherently, bitwise operation) results in the same number it would result in if we directly passed int arrays.

the right approach is to have `usearch_dist` support integer or byte typed arrays.

Reply:
Good point. I can modify `usearch_dist` to take an integer typed array and compute hamming distance appropriately. Let me know